server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1742,7 +1742,7 @@ public void run() throws Exception {
                             // playing safe here and close the engine even if the above succeeds - close can be called multiple times
                             // Also closing refreshListeners to prevent us from accumulating any more listeners
                             IOUtils.close(
-                                engine,
+                                engine != null ? () -> engine.close() : null,
```

Comment:
Could we extract a null-safe utility, say as a static method on `Engine`, rather than sprinkling these null checks across all the callers?